posts/views.py

```diff
@@ -214,6 +215,18 @@ def unlike_post_view(request: HttpRequest, pk: int):
     return redirect(Post.objects.get(pk=pk).get_absolute_url())
 
 
+def share_post_view(request: HttpRequest, pk: int):
+
+    # duplicate the post
+    shared_post = Post.objects.get(pk=pk)
+    shared_post.pk = None
+    print(shared_post)
+    shared_post.save()
+    author = User.objects.get(from_user=request.user)
+    shared_post.author = author
+    return redirect(shared_post.get_absolute_url())
```

Comment:
This'll update the existing post if I recall correctly, you want to manually copy the object by creating a new Post object